common/tools/dev-tool/src/commands/samples/checkNodeVers.ts

```diff
@@ -0,0 +1,323 @@
+        ", "
+      )} docker images but encountered an error doing so`,
+      callback
+    );
+  } else {
+    callback?.();
+  }
+}
+
+function deleteDockerContextCallback(dockerContextDirectory?: string, callback?: () => void) {
+  if (dockerContextDirectory) {
+    log.info(`cleanup: deleting the ${dockerContextDirectory} docker context directory`);
+    spawnCMDIgnoreError("rm", ["-rf", dockerContextDirectory], undefined, callback);
+  } else {
+    callback?.();
+  }
+}
+
+function cleanup(
+  dockerContextDirectory?: string,
+  dockerContainerNames?: string[],
+  dockerImageNames?: string[],
+  callback?: () => void
+) {
+  deleteDockerContextCallback(dockerContextDirectory, () =>
+    deleteDockerContainerCallback(dockerContainerNames, () =>
+      deleteDockerImageCallback(dockerImageNames, callback)
+    )
+  );
+}
+
+function buildRunSamplesScript(
+  containerWorkspacePath: string,
+  packageName: string,
+  envFileName: string,
+  logFilePath?: string
+) {
+  function compileCMD(cmd: string, printToScreen?: boolean) {
+    return printToScreen ? cmd : `${cmd} >> ${logFilePath} 2>&1`;
+  }
+  const printToScreen = logFilePath === undefined;
+  const packagePath = `${containerWorkspacePath}/${packageName}`;
+  const envFilePath = `${containerWorkspacePath}/${envFileName}`;
+  const javascriptSamplesPath = `${containerWorkspacePath}/samples/javascript`;
+  const typescriptCompiledSamplesPath = `${containerWorkspacePath}/samples/typescript/dist`;
+  const scriptContent = `#!/bin/sh
+
+function install_dependencies_helper() {
+\tlocal samples_path=\$1;
+\tcd \${samples_path};
+\t${compileCMD(`npm install ${packagePath}`, printToScreen)}
+\t${compileCMD(`npm install`, printToScreen)}
+}
+
+function install_packages() {
+\techo "Using node \$(node -v) to install dependencies";
+\tinstall_dependencies_helper ${containerWorkspacePath}/samples/javascript
+\tinstall_dependencies_helper ${containerWorkspacePath}/samples/typescript;
+\tcp ${envFilePath} ${containerWorkspacePath}/samples/javascript/;
+}
+
+function run_samples() {
+\tsamples_path=\$1;
+\techo "Using node \$(node -v) to run samples in \${samples_path}";
+\tcd "\${samples_path}";
+\tfor SAMPLE in *.js; do
+\t\tnode \${SAMPLE};
+\tdone
+}
+
+function build_typescript() {
+\techo "Using node \$(node -v) to build the typescript samples";
+\tcd ${containerWorkspacePath}/samples/typescript
+\t${compileCMD(`npm run build`, printToScreen)}
+\tcp ${envFilePath} ${containerWorkspacePath}/samples/typescript/dist/
+}
+
+function main() {
+\tinstall_packages;
+\trun_samples "${javascriptSamplesPath}";
+\tbuild_typescript && run_samples "${typescriptCompiledSamplesPath}";
+}
+
+main`;
+  return scriptContent;
+}
+
+function createDockerContextDirectory(
+  dockerContextDirectory: string,
+  containerWorkspacePath: string,
+  samples_path: string,
+  envPath: string,
+  packagePath?: string,
+  logFilePath?: string,
+  callback?: () => void
+): void {
+  if (packagePath === undefined) {
```

Comment:
Should we make it required in the function arguments? And, if someone passes in an empty string this would not throw - maybe `if (!packagePath)`

Reply:
I decided to put the validation here to decrease the logic in the main function. Yeah it is expected the path you pass to be valid. I will add a check for existence of the file.